core/src/subgraph/instance.rs

```diff
@@ -188,9 +188,9 @@ where
     pub(crate) fn add_dynamic_data_source(
         &mut self,
         logger: &Logger,
-        data_source: C::DataSource,
-        templates: Arc<Vec<C::DataSourceTemplate>>,
-        metrics: Arc<HostMetrics>,
+        data_source: &C::DataSource,
```

Comment:
if these are going to be cloned further down maybe it's worth letting the caller clone it and in this case I think you could just pass the ownership down in line 212

Reply:
Great suggestion! Just fixed on 971275892d915cc5361454fa92851dde42163a17